src/core/qgsproject.cpp

```diff
@@ -410,30 +410,34 @@ QgsProject::~QgsProject()
 {
   mIsBeingDeleted = true;
 
-  clear();
+  if ( this != sProject.get() )
```

Comment:
Why isn't this done on sProject? This is what was leading to the earlier crashes -- there's a specific tear-down order in clear() which needs to be followed here.

Reply:
because it leads to a reference to QgsSettings after the application is already gone and crashed.  QgsProject was never destroyed in app and crssync leaving the temporary files around.

Reply:
In that case the tear down order from clear() will need to be copied here too

Reply:
> because it leads to a reference to QgsSettings after the application is already gone and crashed. QgsProject was never destroyed in app and crssync leaving the temporary files around.

correction: the project was actually destroyed - but another one was created when the QgsProviderRegistry tried to remove "all" of the "current" project's layers - an that one was never destroyed.